KeyVaultCA/Program.cs

```diff
@@ -9,52 +9,61 @@ namespace KeyVaultCA
-            public string KeyVaultUrl { get; set; }
+        //    [Option("kvUrl", Required = true, HelpText = "Key Vault URL")]
+        //    public string KeyVaultUrl { get; set; }
 
-            // Certificates
+        //    // Certificates
 
-            [Option("issuercert", Required = true, HelpText = "Name of the issuing certificate in KeyVault.")]
-            public string IssuerCertName { get; set; }
+        //    [Option("issuercert", Required = true, HelpText = "Name of the issuing certificate in KeyVault.")]
+        //    public string IssuerCertName { get; set; }
 
-            // Options for the end entity certificate
+        //    // Options for the end entity certificate
 
-            [Option("csrPath", Required = false, HelpText = "Path to the CSR file in .der format")]
-            public string PathToCsr { get; set; }
+        //    [Option("csrPath", Required = false, HelpText = "Path to the CSR file in .der format")]
+        //    public string PathToCsr { get; set; }
 
-            [Option("output", Required = false, HelpText = "Output file name for the certificate")]
-            public string OutputFileName { get; set; }
+        //    [Option("output", Required = false, HelpText = "Output file name for the certificate")]
+        //    public string OutputFileName { get; set; }
 
-            // Options for Root CA creation
+        //    // Options for Root CA creation
 
-            [Option("ca", Required = false, HelpText = "Should register Root CA")]
-            public bool IsRootCA { get; set; }
+        //    [Option("ca", Required = false, HelpText = "Should register Root CA")]
+        //    public bool IsRootCA { get; set; }
 
-            [Option("subject", Required = false, HelpText = "Subject in the format 'C=US, ST=WA, L=Redmond, O=Contoso, OU=Contoso HR, CN=Contoso Inc'")]
-            public string Subject { get; set; }
-        }
+        //    [Option("subject", Required = false, HelpText = "Subject in the format 'C=US, ST=WA, L=Redmond, O=Contoso, OU=Contoso HR, CN=Contoso Inc'")]
+        //    public string Subject { get; set; }
+        //}
 
-        static void Main(string[] args)
+        static async Task Main(string[] args)
         {
-            Parser.Default.ParseArguments<Options>(args)
-                   .WithParsed(o =>
-                   {
-                       StartAsync(o).Wait();
-                   });
+            IConfiguration config = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: true)
+                .AddCommandLine(args)
+                .AddEnvironmentVariables()
+                .Build();
+
+            await CreateCertificate(config);
         }
 
-        private static async Task StartAsync(Options o)
+        private static async Task CreateCertificate(IConfiguration config)
         {
+            var estConfig = new EstConfiguration();
```

Comment:
you can also go for the slightly more concise 
`var estConfig = Configuration.GetSection("KeyVault").Get<EstConfiguration>();`